libimage/pull_test.go

```diff
@@ -122,3 +122,31 @@ func TestPullPlatforms(t *testing.T) {
 	require.NoError(t, err, "lookup busybox - by arm")
 	require.NotNil(t, image, "lookup busybox - by local arch")
 }
+
+func TestPullPolicy(t *testing.T) {
+	runtime, cleanup := testNewRuntime(t)
+	defer cleanup()
+	ctx := context.Background()
+	pullOptions := &PullOptions{}
+
+	pulledImages, err := runtime.Pull(ctx, "alpine", config.PullPolicyAlways, pullOptions)
+	require.NoError(t, err, "pull alpine")
+	require.Len(t, pulledImages, 1)
+
+	if goruntime.GOARCH != "arm" {
+		pullOptions.Architecture = "arm"
+	} else {
+		pullOptions.Architecture = "amd64"
+	}
+	pulledImages, err = runtime.Pull(ctx, "alpine", config.PullPolicyNever, pullOptions)
+	require.Error(t, err, "Never pull different arch alpine")
```

Comment:
This is failing because we're still ignoring the arch for image lookup, see https://github.com/containers/common/blob/main/libimage/pull.go#L432.

We also need to touch the local lookup and set the arch/os/variant if it's `!= ""` in the lookup options, see https://github.com/containers/common/blob/ab2131e65fee259b81089684d4e5eddecfcdec68/libimage/runtime.go#L169-L175.

Warning: there are many images who get the arch/variant wrong, so doing the platform checks may return false positives/negatives.